### PR TITLE
Upgrade quarkus-arquillian to use JUnit 5

### LINKED ISSUE #27812: Upgrade quarkus-arquillian to use JUnit 5
### Description

The quarkus-arquillian extension is still using the old JUnit 4 framework.
Most usages are under the tcks.
It's getting outdated because It seems that MP is trying to move towards Junit, starting that after MP 6.0.
Also, MP Telemetry tracing is based on JUnit 5.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
Fix https://github.com/quarkusio/quarkus/issues/27812

### CODE REVIEW

**Comment by Sgitario** [comment]
If you decide to proceed with the changes in this pull request, it should be noted in the migration guide as a breaking change.

**Comment by gsmet** [comment]
@Sgitario if /we/ decide, you're part of the team.

**Comment by gsmet** [comment]
@mkouba did you remember why you targeted JUnit 4? Was it because of a lack of support of JUnit 5 in Arquillian or did we need it for some TCKs?

**Comment by mkouba** [comment]
> @mkouba did you remember why you targeted JUnit 4? Was it because of a lack of support of JUnit 5 in Arquillian or did we need it for some TCKs?

It was the lack of support in Arquillian. And I believe that the TCKs that used JUnit at that time were using JUnit4 APIs.

File: tcks/microprofile-metrics/api/pom.xml [hunk 1]
```diff
@@ -39,6 +39,9 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arquillian</artifactId>
+            <exclusions>
+
+            </exclusions>
```
Comment (gsmet) [inline tcks/microprofile-metrics/api/pom.xml:44]: This should go away.

File: test-framework/arquillian/src/test/java/io/quarkus/arquillian/test/SimpleTest.java [hunk 2]
```diff
@@ -10,16 +10,19 @@
...
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ArquillianExtension.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (gsmet) [inline test-framework/arquillian/src/test/java/io/quarkus/arquillian/test/SimpleTest.java:25]: You rewrote all the tests for JUnit 5. Should we at least duplicate one for JUnit 4 if it's possible to have both around?

File: test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusJunitCallbacks.java [hunk 3]
```diff
@@ -18,25 +15,27 @@
  */
 class QuarkusJunitCallbacks {
```
Comment (gsmet) [inline test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusJunitCallbacks.java:16]: Should this class be abstract now?

**Review by gsmet — CHANGES_REQUESTED** [review]
OK, I think we're not far from being able to merge this PR.

I added a few comments. As for testing JUnit 4, it's either simple and we do it or it's not and we can live without it. Don't spend time on that.

File: test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusJunitCallbacks.java (hunk 3 above)
Comment (Sgitario) [reply test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusJunitCallbacks.java:16]: +1

File: test-framework/arquillian/src/test/java/io/quarkus/arquillian/test/SimpleTest.java (hunk 2 above)
Comment (Sgitario) [reply test-framework/arquillian/src/test/java/io/quarkus/arquillian/test/SimpleTest.java:25]: Unfortunately, unless we do some tricky stuff, we can't instruct the surefire plugin to run both junit 5 and junit 4 tests. 
Anyway, all the tcks modules are using the JUnit 4, so we have both covered already.

**Comment by Sgitario** [comment]
PR updated with the resolved change requests.

File: test-framework/arquillian/src/test/java/io/quarkus/arquillian/test/SimpleTest.java (hunk 2 above)
Comment (gsmet) [reply test-framework/arquillian/src/test/java/io/quarkus/arquillian/test/SimpleTest.java:25]: Yeah, fine by me. Thanks!

**Review by gsmet — APPROVED** [review]
Thanks. Nice having this ready for future MP versions.

**Comment by gsmet** [comment]
@radcortez FYI regarding MP TCKs.

**Pull request merged**